Exclude OCSP sites for BOD 18-01 compliance calculations

jsf9k opened the pull request:
This pull request:
- Excludes OCSP sites (as listed [here](https://github.com/GSA/data/blob/master/dotgov-websites/ocsp-crl.csv)) from BOD 18-01 compliance calculations
- Still lists OCSP domains in the grand table in the HTTPS report.  Their names are followed by an asterisk to indicate they are special.
- Adds a blurb to the report text that explains this.
- Adds a boolean field to the CSV attachment indicating whether the domain is an OCSP domain or not.

This code has been tested in a full staging deployment on AWS, and it appears to be working.

jsf9k/cyhy-reports#13 is a corresponding pull request that goes along with this one, keeping the CybEx scorecard in sync with the HTTPS reports.

jsf9k committed:
Fix a bunch of minor but annoying style issues

flake8, for instance, will complain vehemently about these.

jsf9k committed:
Add code to pull down the OCSP exclusion list

jsf9k committed:
More minor formatting fixes

jsf9k committed:
Bumped version from 1.0.8 to 1.1.0

jsf9k committed:
Exclude OCSP domains from BOD compliance calculations

Such domains are still scored and presented in the report.

jsf9k committed:
Include OCSP domains in the table, but denote they are special

An asterisk following the domain is used to denote them.  Also add a
blurb explaining this to the report recipient.

dav3r commented on report/https_scan_report.mustache:
```diff
@@ -427,10 +437,10 @@ BOD 18-01 includes \href{https://https.cio.gov/guide/#what-about-domains-that-ar
-<<#live_bool>>\color{dhs-blue}\textbf{{<<&domain>>}} & <<#domain_supports_https_bool>>\checkmark <</domain_supports_https_bool>><<^domain_supports_https_bool>> \xmark <<#https_bad_chain_bool>>\textit{\newline Bad chain}<</https_bad_chain_bool>><<#https_bad_hostname_bool>>\textit{\newline Bad hostname}<</https_bad_hostname_bool>><<#https_expired_cert_bool>>\textit{\newline Expired certificate}<</h ...
-<<^live_bool>>\color{dhs-blue}\textbf{{<<&domain>>}} \textit{\ \ Not Live} & \textit{N/A} & \textit{N/A} & \textit{N/A} & \textit{N/A} & \checkmark & <<#hsts_preloaded_bool>> \checkmark <</hsts_preloaded_bool>><<^hsts_preloaded_bool>><<#hsts_preload_pending_bool>> \textit{Pending*} <</hsts_preload_pending_bool>><<^hsts_preload_pending_bool>><<#hsts_preload_ready_bool>> \textit{Preload-ready*} <</ ...
+<<#live_bool>>\color{dhs-blue}\textbf{{<<&domain>><<#ocsp_domain>>*<</oscp_domain>>}} & <<#domain_supports_https_bool>>\checkmark <</domain_supports_https_bool>><<^domain_supports_https_bool>> \xmark <<#https_bad_chain_bool>>\textit{\newline Bad chain}<</https_bad_chain_bool>><<#https_bad_hostname_bool>>\textit{\newline Bad hostname}<</https_bad_hostname_bool>><<#https_expired_cert_bool>>\textit{ ...
+<<^live_bool>>\color{dhs-blue}\textbf{{<<&domain>><<#ocsp_domain>>*<</oscp_domain>>}} \textit{\ \ Not Live} & \textit{N/A} & \textit{N/A} & \textit{N/A} & \textit{N/A} & \checkmark & <<#hsts_preloaded_bool>> \checkmark <</hsts_preloaded_bool>><<^hsts_preloaded_bool>><<#hsts_preload_pending_bool>> \textit{Pending*} <</hsts_preload_pending_bool>><<^hsts_preload_pending_bool>><<#hsts_preload_ready_b ...
```
Typo here:
```<<#ocsp_domain>>``` should be ```<<#oscp_domain>>```
(o**cs**p vs. o**sc**p)

jsf9k replied:
It's the other way around.  `<</oscp_domain>>` should be `<</ocsp_domain>>`.

dav3r replied:
LOL, that's clearly what I meant ;)

jsf9k replied:
Thanks for pointing it out.  It would have taken me some time to find that.

dav3r commented on report/https_scan_report.mustache:
```diff
@@ -427,10 +437,10 @@ BOD 18-01 includes \href{https://https.cio.gov/guide/#what-about-domains-that-ar
+<<^live_bool>>\color{dhs-blue}\textbf{{<<&domain>><<#ocsp_domain>>*<</oscp_domain>>}} \textit{\ \ Not Live} & \textit{N/A} & \textit{N/A} & \textit{N/A} & \textit{N/A} & \checkmark & <<#hsts_preloaded_bool>> \checkmark <</hsts_preloaded_bool>><<^hsts_preloaded_bool>><<#hsts_preload_pending_bool>> \textit{Pending*} <</hsts_preload_pending_bool>><<^hsts_preload_pending_bool>><<#hsts_preload_ready_b ...
 % Show subdomains if hsts_preloaded_bool is false and if any subdomain_scores exist
-<<^hsts_preloaded_bool>><<#subdomain_scores>>\color{dhs-blue}{\ \ \ <<&domain>>} & <<#domain_supports_https_bool>>\checkmark <</domain_supports_https_bool>><<^domain_supports_https_bool>> \xmark <<#https_bad_chain_bool>>\textit{\newline Bad chain}<</https_bad_chain_bool>><<#https_bad_hostname_bool>>\textit{\newline Bad hostname}<</https_bad_hostname_bool>><<#https_expired_cert_bool>>\textit{\newl ...
+<<^hsts_preloaded_bool>><<#subdomain_scores>>\color{dhs-blue}{\ \ \ <<&domain>><<#ocsp_domain>>*<</oscp_domain>>} & <<#domain_supports_https_bool>>\checkmark <</domain_supports_https_bool>><<^domain_supports_https_bool>> \xmark <<#https_bad_chain_bool>>\textit{\newline Bad chain}<</https_bad_chain_bool>><<#https_bad_hostname_bool>>\textit{\newline Bad hostname}<</https_bad_hostname_bool>><<#https ...
```
Same typo here:
```<<#ocsp_domain>>``` should be ```<<#oscp_domain>>```
(o**cs**p vs. o**sc**p)

dav3r commented:
@jsf9k, this approach makes sense to me.

jsf9k committed:
Fix a couple of typos

jsf9k committed:
Use None instead of 0 for the dummy dict value

This may save a little memory and yields the same result.

jsf9k committed:
Use a with clause to ensure file handle is cleaned up

jsf9k committed:
Fix two bugs

Point to the correct location for the OCSP exclusion file, and add a
line setting ocsp_exclusion for subdomains (just like it is already
being set for base domains).

jsf9k committed:
Fix bug where subdomain value for ocsp_domain was incorrect

The value for the base domain was incorrectly being used.

jsf9k committed:
Cosmetic changes in information printed to the screen

jsf9k committed:
Add some comments

These comments explain why we don't exclude OCSP domains when
performing the queries for self.__domain_count and
self.__base_domain_count.

dav3r reviewed (approved):
Super strong work @jsf9k !  💪 💪 💪
